core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -120,6 +120,12 @@ public void setConf(Configuration conf) {
 
   @Override
   public Configuration getConf() {
+    // Create a default hadoopConf as it is required for the object to be valid.
+    // E.g. newInputFile would throw NPE with getConf() otherwise.
+    if (hadoopConf == null) {
+      this.hadoopConf = new SerializableConfiguration(new Configuration())::get;
```

Comment:
Is there a potential for a race condition here? Do you see a need to add synchronization to protect this instantiation?

Reply:
yeah, `FileIO` can potentially be used by multiple threads. let me add synchronization.